perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_resource_attribute_def_virt_isBanned.java

```diff
@@ -62,6 +149,29 @@ public AttributeDefinition getAttributeDefinition() {
         return attr;
     }
 
-
+	/**
+	 * Get List of pairs of Resource and Member affected by change in Ban (from user and facility)
+	 *
+	 * @param perunSession
+	 * @param user affected user
+	 * @param facility affected facility
+	 * @return list of pairs of REsource and Members affected by Ban
+	 * @throws InternalErrorException
+	 */
+	private List<Pair<Resource, Member>> getAffectedMemberResourceObjects(PerunSessionImpl perunSession, User user, Facility facility) throws InternalErrorException {
+		List<Pair<Resource, Member>> listOfAffectedObjects = new ArrayList<>();
+		List<Resource> assignedResources = perunSession.getPerunBl().getFacilitiesManagerBl().getAssignedResources(perunSession, facility);
+		List<Member> userMembers = perunSession.getPerunBl().getMembersManagerBl().getMembersByUser(perunSession, user);
```

Comment:
You don't have to retrieve all user members, since Member contains ``userId``, which you can check against assigned members for each resource.